src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java

```diff
@@ -0,0 +1,97 @@
+package de.themorpheus.edu.taskservice.controller.solution;
+
+import de.themorpheus.edu.taskservice.database.model.solution.SolutionModel;
+import de.themorpheus.edu.taskservice.database.model.solution.FreestyleSolutionModel;
+import de.themorpheus.edu.taskservice.database.repository.solution.SolutionFreestyleRepository;
+import de.themorpheus.edu.taskservice.endpoint.dto.request.solution.CheckFreestyleSolutionRequestDTO;
+import de.themorpheus.edu.taskservice.endpoint.dto.request.solution.CreateFreestyleSolutionRequestDTO;
+import de.themorpheus.edu.taskservice.endpoint.dto.request.solution.UpdateFreestyleSolutionRequestDTO;
+import de.themorpheus.edu.taskservice.endpoint.dto.response.solution.GetFreestyleSolutionResponseDTO;
+import de.themorpheus.edu.taskservice.util.ControllerResult;
+import de.themorpheus.edu.taskservice.util.Error;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static de.themorpheus.edu.taskservice.util.Constants.Solution.Freestyle.NAME_KEY;
+
+@Component
+public class FreestyleSolutionController implements Solution {
+
+	@Autowired
+	private SolutionFreestyleRepository solutionFreestyleRepository;
+
+	@Autowired
+	private SolutionController solutionController;
+
+	public ControllerResult<FreestyleSolutionModel> createSolutionFreestyle(CreateFreestyleSolutionRequestDTO dto) {
+		ControllerResult<SolutionModel> solutionResult = this.solutionController.getOrCreateSolution(dto.getTaskId(), NAME_KEY);
+		if (solutionResult.isResultNotPresent()) return ControllerResult.ret(solutionResult);
+
+		if (this.solutionFreestyleRepository.existsById(dto.getTaskId()))
+			return ControllerResult.of(Error.ALREADY_EXISTS, NAME_KEY);
+
+		return ControllerResult.of(this.solutionFreestyleRepository.save(
+				new FreestyleSolutionModel(-1, solutionResult.getResult(), dto.getSolution())
+				)
+		);
+	}
+
+	public ControllerResult<FreestyleSolutionModel> checkSolutionFreestyle(CheckFreestyleSolutionRequestDTO dto) {
+		ControllerResult<SolutionModel> solutionResult = this.solutionController.getGenericSolution(dto.getTaskId(), NAME_KEY);
+		if (solutionResult.isResultNotPresent()) return ControllerResult.ret(solutionResult);
+
+		Optional<FreestyleSolutionModel> optionalFreestyleSolution = this.solutionFreestyleRepository.findById(solutionResult.getResult().getSolutionId());
+		if (!optionalFreestyleSolution.isPresent()) return ControllerResult.of(Error.NOT_FOUND, NAME_KEY);
+
+		if (!optionalFreestyleSolution.get().getSolution().equalsIgnoreCase(dto.getSolution()))
+			return ControllerResult.of(Error.WRONG_ANSWER, NAME_KEY);
+
+		return ControllerResult.empty();
```

Comment:
return solution

Reply:
create a new Model and a new table to save the solution of a user and the teacher can access them
UUID is given from the Endpoint but work in progress, so use UUID.random

Reply:
same for Image

Reply:
Also please create an issue for creating methods that a teacher can get all solutions the user sent to a specific task

Reply:
> create a new Model and a new table to save the solution of a user and the teacher can access them
> UUID is given from the Endpoint but work in progress, so use UUID.random

How should I make this?

Reply:
Make A "UserFreestyleSolutionModel" and a "UserFreestyleSolutionRepository" like the other Repository classes and Models and save the solutions the user sent